Userland/Libraries/LibWeb/UIEvents/WheelEvent.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibGUI/Event.h>
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/HTML/EventNames.h>
+#include <LibWeb/UIEvents/EventNames.h>
+#include <LibWeb/UIEvents/WheelEvent.h>
+
+namespace Web::UIEvents {
+
+WheelEvent::WheelEvent(JS::Realm& realm, FlyString const& event_name, WheelEventInit const& event_init)
+    : MouseEvent(realm, event_name, event_init)
+    , m_delta_x(event_init.delta_x)
+    , m_delta_y(event_init.delta_y)
+    , m_delta_mode(event_init.delta_mode)
+{
+    set_prototype(&Bindings::cached_web_prototype(realm, "WheelEvent"));
+    set_event_characteristics();
+}
+
+WheelEvent::~WheelEvent() = default;
+
+WheelEvent* WheelEvent::create(JS::Realm& realm, FlyString const& event_name, WheelEventInit const& event_init)
+{
+    return realm.heap().allocate<WheelEvent>(realm, realm, event_name, event_init);
+}
+
+WheelEvent* WheelEvent::create_from_platform_event(JS::Realm& realm, FlyString const& event_name, double offset_x, double offset_y, double client_x, double client_y, double delta_x, double delta_y)
+{
+    WheelEventInit event_init {};
+    event_init.offset_x = offset_x;
+    event_init.offset_y = offset_y;
+    event_init.client_x = client_x;
+    event_init.client_y = client_y;
+    event_init.delta_x = delta_x;
+    event_init.delta_y = delta_y;
+    event_init.delta_mode = 0;
+    return WheelEvent::create(realm, event_name, event_init);
+}
+
+void WheelEvent::set_event_characteristics()
+{
+    if (type().is_one_of(EventNames::wheel)) {
```

Comment:
Not sure we need the `is_one_of` check here. There's only one kind of WheelEvent,